src/xmerl_dsig.erl

```diff
@@ -65,7 +65,7 @@ sign(ElementIn, PrivateKey = #'RSAPrivateKey'{}, CertBin, SigMethod) when is_bin
             case lists:keyfind('id', 2, ElementStrip#xmlElement.attributes) of
                 #xmlAttribute{value = LowId} -> {ElementStrip, LowId};
                 _ ->
-                    NewId = uuid_drv:uuid_hex(),
+                    NewId = "YOYO" ++ uuid_drv:uuid_hex(),
```

Comment:
I found in some scenarios ADFS will reject the SAM request if the request id starts with a number. 
This line is to hack the limitation of ADFS.